pkg/mdformatter/mdformatter.go

```diff
@@ -109,11 +110,31 @@ func (FormatFrontMatter) TransformFrontMatter(_ SourceContext, frontMatter map[s
 	}
 	sort.Sort(sort.Reverse(sort.StringSlice(keys)))
 
-	b := bytes.NewBuffer([]byte("---\n"))
+	b := bytes.NewBuffer([]byte("---"))
 	for _, k := range keys {
-		_, _ = fmt.Fprintf(b, "%v: %v\n", k, frontMatter[k])
+		// check if frontMatter[k] is a map
+		frontMatterStr := fmt.Sprintf("%s", frontMatter[k])
+		isMap := strings.HasPrefix(frontMatterStr, "map[")
+
+		// split string to get all key-value pairs
+		if isMap {
+			_, _ = fmt.Fprintf(b, "\n%v:", k)
+			pairs := strings.Split(frontMatterStr[4:len(frontMatterStr)-1], " ")
+			pair := make([]string, 2)
+			// ignore spaces in values
+			for _, val := range pairs {
+				if strings.Contains(val, ":") {
+					pair = strings.Split(val, ":")
+					_, _ = fmt.Fprintf(b, "\n  %v: %v", pair[0], pair[1])
+				} else {
```

Comment:
We can improve this: https://thanos.io/tip/contributing/coding-style-guide.md/#control-structure-prefer-early-returns-and-avoid-else 🤗

Reply:
We can avoid else here

Reply:
Alright